src/averageAges.js

```diff
@@ -55,7 +70,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenWithChil = people
+    .filter(child => child.mother && people
+      .find(person => person.name === child.mother));
+
+  const withSonOnly = womenWithChil
+    .filter(person => person.sex === 'm');
+
+  const ageDifference = array => array.reduce((sum, x) =>
+    sum + x.born - people
+      .find(person => person.name === x.mother).born, 0) / array.length;
```

Comment:
What is `x`?

Reply:
again, Mykola is right. Your variable names should be maximally clear. The variable name gives us a clear understanding of what is stored inside
You should never use names like `x`, they don't give any info

Reply:
Thanks. Probably I left it from my draft code... Fixed it!